### DISCUSSION TITLE
Prado framework forum

### DISCUSSION THREAD

**Discussion by RobCrack2023** [pradosoft/prado#931] [General]
Hello everyone, I have the intention of creating a prado framework forum, and I wanted to know your opinion first and also if I can count on you to help me respond to users who have problems or doubts with the code, thank you very much in advance

**Comment by belisoful** [comment]
I'd love to see such a thing but debugging other peoples' code when I have so much to debug myself is a stretch.  I'll answer general questions. I  was actually thinking of doing something similar.  Are you going to use an existing framework for the forum?  or are you going to build a PRADO based forum?

Would you be interested in making the forum code into its own Composer package?  THAT is something I'd LOVE to see.

There has been some advances since the published PRADO documentation was last updated, and isn't documented on the site officially:
- Web Control Decorations (Skinning via theme to decorate controls with display HTML)
- Behaviors (regular and class) (object and class-wide).  class behaviors can attach to multiple owners. regular behaviors only attach to one.   class-wide behaviors attach to an entire class and all subclasses.  class-wide behaviors can also be installed on Interfaces and first level traits.
- Dynamic 'dy' events between behaviors and their owners. 
- global 'fx' events at the system level.  (used by Prado internally to give existing instances new class-wide behaviors.)
- Cron
- RBAC permissions
- New Command Shell (with optional system login), color support
- Composer integration

**Reply by RobCrack2023** [reply]
Well, my idea at first will be to use a framework for forums, later I will build it with prado, but for now it is to make prado known, I must say that years ago I used it a lot and the forum that existed helped beginners a lot, When I have it set up I will let you know and I know you are quite busy but your advice will always be well received. I must also mention that my official language is Spanish, so the forum will be for two languages, firstly Spanish and English, I am letting you know progress, regards

**Comment by belisoful** [comment]
The community aspect of PRADO is lacking in the recent years.

**Comment by ctrlaltca** [comment]
Yep, unfortunately the loss of the old website and forum has been a hard hit.

**Comment by belisoful** [comment]
can we resurrect the old site?  what will it take for a new site?  (as in depth as you would like.)

**Comment by ctrlaltca** [comment]
The old website was run by robin, but he just disappeared from the internet one day in 2015.
I tried to contact qiang to get at least the old pradosoft.com domain, but he didn't even reply and let the domain expire and get squatted.
Thus the new domain pradoframework.net, the "site" in prado-demos, and the change from "pradosoft" to "the prado group" to dissociate from the squatters.
Not even internet archive contains a full copy of the old forum: https://web.archive.org/web/20150206032829/http://www.pradosoft.com/forum/

**Comment by belisoful** [comment]
I have a web space that can be used.  I was hoping there would be more composer PRADO plugins by now.  I think my google analytics basic plugin is the first and still only plugin.  I have ideas to make for plugins but the core needs work first.

A website with a forum, listed composer plugins, and documentation would do wonders.
